### PR TITLE
Test varying DB max conns

### PR DESCRIPTION
**Pull request by kegsay**
Typically the failure mode is where some request needs 2+ txns by accident, which causes problems if that then blocks other things from making forward progress, causing deadlocks.

### CODE REVIEW

File: v3.go [hunk 1]
```diff
@@ -75,6 +79,15 @@ func Setup(destHomeserver, postgresURI, secret string, opts Opts) (*handler2.Han
 	}
 	store := state.NewStorage(postgresURI)
 	storev2 := sync2.NewStore(postgresURI, secret)
+	for _, db := range []*sqlx.DB{store.DB, storev2.DB} {
+		if opts.DBMaxConns > 0 {
+			db.SetMaxOpenConns(opts.DBMaxConns)
+			db.SetMaxIdleConns(opts.DBMaxConns)
```
Comment (DMRobertson) [inline v3.go:85]: Think you said there was some mysql driver that advised doing this. Can you drop a link here for our future selves?

File: tests-integration/db_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			m.MatchResponse(t, res, m.MatchRoomSubscriptionsStrict(map[string][]m.RoomMatcher{
+				roomID: {
+					m.MatchRoomTimelineMostRecent(1, []json.RawMessage{dripMsg}),
+				},
+			}))
+			t.Logf("user %s has received the drip", userID)
+		}(i)
+	}
+
+	wg.Wait()
```
Comment (DMRobertson) [inline tests-integration/db_test.go]: Does this have a timeout attached to it? (I.e. will this test always promptly terminate in CI?)

Comment (kegsay) [reply tests-integration/db_test.go]: No, it'll rely on `go test -timeout`

**Review by DMRobertson — APPROVED** [review]

**Comment by DMRobertson** [comment]
(Isn't it worth committing this?)

**Comment by DMRobertson** [comment]
(Oh, you merged manually rather than via github)

**Pull request closed without merging**